File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int maxAreaPortion = 0;

        int i = 0;
        int j = heights.size() -1;

        while(i<j){
            maxAreaPortion = max(maxAreaPortion, (j-i)*(min(heights[i], heights[j])));

            if(heights[i] < heights[j]){
                i++;
            }else if(heights[i] > heights[j]){
                j--;
            }else{
                i++;
                j--;
            }
        }

        return maxAreaPortion;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.js
class Solution {
    /**
     * @param {number[]} prices
     * @return {number}
     */
    maxProfit(prices) {
        let max_profit = 0;
        let min_price = Number.MAX_VALUE;

        for(let price of prices){
            min_price = Math.min(min_price, price);
            max_profit = Math.max(max_profit, price - min_price);
        }

        return max_profit;
    }
}
